extmod/modbluetooth.c

```diff
@@ -1022,12 +1025,29 @@ void mp_bluetooth_gattc_on_descriptor_result(uint16_t conn_handle, uint16_t hand
     MICROPY_PY_BLUETOOTH_EXIT
 }
 
+void mp_bluetooth_gattc_read_error(uint8_t error, uint16_t conn_handle, uint16_t value_handle) {
+    MICROPY_PY_BLUETOOTH_ENTER
+    mp_obj_bluetooth_ble_t *o = MP_OBJ_TO_PTR(MP_STATE_VM(bluetooth));
+    if (enqueue_irq(o, 2 + 2 + 1 + 1, MP_BLUETOOTH_IRQ_GATTC_READ_RESULT)) {
+        ringbuf_put16(&o->ringbuf, conn_handle);
+        ringbuf_put16(&o->ringbuf, value_handle);
+        ringbuf_put(&o->ringbuf, error);
+        ringbuf_put(&o->ringbuf, 0 /* data_len */);
+    }
+    schedule_ringbuf();
+    MICROPY_PY_BLUETOOTH_EXIT
+}
+
 size_t mp_bluetooth_gattc_on_data_available_start(uint16_t event, uint16_t conn_handle, uint16_t value_handle, size_t data_len) {
     mp_obj_bluetooth_ble_t *o = MP_OBJ_TO_PTR(MP_STATE_VM(bluetooth));
+    size_t error_len = (event == MP_BLUETOOTH_IRQ_GATTC_READ_RESULT) ? 1 : 0;
     data_len = MIN(o->irq_data_data_alloc, data_len);
-    if (enqueue_irq(o, 2 + 2 + 1 + data_len, event)) {
+    if (enqueue_irq(o, 2 + 2 + 1 + error_len + data_len, event)) {
```

Comment:
This calculates the length of the ring buffer. I guess since error is a byte this should be just 1 here?

Reply:
The function is also used for `MP_BLUETOOTH_IRQ_GATTC_NOTIFY` and `MP_BLUETOOTH_IRQ_GATTC_INDICATE`, where adding the status code probably does not make sense...?

Reply:
Oh I see, you are right yes, sorry about that.